src/services/createCallbackContext.ts

```diff
@@ -0,0 +1,34 @@
+import { RouterPushError } from '@/errors/routerPushError'
+import { RouterRejectionError } from '@/errors/routerRejectionError'
+import { RegisteredRouterPush, RegisteredRouterReject, RegisteredRouterReplace } from '@/types/register'
+import { RouterPushOptions } from '@/types/routerPush'
+import { isUrl } from '@/types/url'
+
+export type CallbackContext = {
+  reject: RegisteredRouterReject,
+  push: RegisteredRouterPush,
+  replace: RegisteredRouterReplace,
```

Comment:
We should update the return types for these to `never` that way typescript will correctly infer that they always throw an error. Then the `throw` in implementation won't be necessary for type guarding or highlighting unreachable code.